iroh/src/magicsock/transports/relay.rs

```diff
@@ -100,6 +100,19 @@ impl RelayTransport {
                 }
             };
 
+            if buf_out.len() < dm.datagrams.contents.len() {
+                // Our receive buffer isn't big enough to process this datagram.
+                // Continuing would cause a panic.
+                break;
```

Comment:
maybe we should log this? especially to make sure we appropriately split things now?